www/src/components/layout/WithNotifications.js

```diff
@@ -0,0 +1,191 @@
+import { useState } from 'react'
+import { Link } from 'react-router-dom'
+import { Avatar, Div, Flex, Img, P } from 'honorable'
+import { CloseIcon } from 'pluralsh-design-system'
+import truncate from 'lodash.truncate'
+import moment from 'moment'
+
+import usePaginatedQuery from '../../hooks/usePaginatedQuery'
+
+import InfiniteScroller from '../utils/InfiniteScroller'
+import Markdown from '../incidents/Markdown'
+
+import { NOTIFICATIONS_QUERY } from './queries'
+
+function WithNotifications({ children }) {
+  const [notifications, loadingNotifications, hasMoreNotifications, fetchMoreNotifications] = usePaginatedQuery(
+    NOTIFICATIONS_QUERY,
+    { variables: {} },
+    data => data.notifications
+  )
+  const [isPanelOpen, setIsPanelOpen] = useState(false)
+
+  return (
+    <Flex>
+      {children({
+        notificationsCount: notifications.length,
+        toggleNotificationsPanel: () => setIsPanelOpen(x => !x),
+      })}
+      {isPanelOpen && (
+        <Flex
+          direction="column"
+          borderRight="1px solid border"
+          width={512}
+          maxWidth={512}
+        >
+          <Flex
+            align="center"
+            px={1}
+            height={57}
+            flexShrink={0}
+            borderBottom="1px solid border"
+          >
+            <P ml={1}>
+              Notifications
+            </P>
+            <Div flexGrow={1} />
+            <Flex
+              p={0.5}
+              mr={-0.25}
+              align="center"
+              justify="center"
+              hoverIndicator="background-light"
+              cursor="pointer"
+              borderRadius="50%"
+              onClick={() => setIsPanelOpen(false)}
+            >
+              <CloseIcon size={12} />
+            </Flex>
+          </Flex>
+          <Flex
+            direction="column"
+            flexGrow={1}
+          >
+            <InfiniteScroller
+              loading={loadingNotifications}
+              hasMore={hasMoreNotifications}
+              loadMore={fetchMoreNotifications}
+              // Allow for scrolling in a flexbox layout
+              flexGrow={1}
+              height={0}
+            >
+              {notifications.map(notification => (
+                <Notification
+                  key={notification.id}
+                  notification={notification}
+                  closePanel={() => setIsPanelOpen(false)}
+                />
+              ))}
+            </InfiniteScroller>
+          </Flex>
+        </Flex>
+      )}
+    </Flex>
+  )
+}
+
+function Notification({ notification, closePanel }) {
+  console.log('notification', notification)
+  const { actor, type, repository, incident, insertedAt } = notification
+
+  function getUrl() {
```

Comment:
why is this not just `const url = type === 'LOCKED' ? '/repository/${repository.id}` : ...`

Reply:
Haha I copy pasted your code 😛